api/desecapi/crypto.py

```diff
@@ -0,0 +1,37 @@
+from base64 import urlsafe_b64encode
+from enum import Enum
+
+from cryptography.fernet import Fernet, InvalidToken
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.kbkdf import CounterLocation, KBKDFHMAC, Mode
+from cryptography.hazmat.backends import default_backend
+from django.conf import settings
+from django.utils.encoding import force_bytes
+
+
+def _derive_urlsafe_key(*, label, context):
+    backend = default_backend()
+    kdf = KBKDFHMAC(algorithm=hashes.SHA256(), mode=Mode.CounterMode, length=32, rlen=4, llen=4,
```

Comment:
From the docs I see that this can fail already. Probably won't happen with all the defaults used here, but if it does, how do you handle that?

Reply:
Currently, uncaught exceptions are turned into 500 errors which in turn leads to an email getting sent to us with details, so that we can track down what happened. So far this has served us well in finding unforeseen edge cases.

If this would happen here, it would certainly be a server-side error (5xx), and we would certainly want to know about it (and not silence it), so I figured this approach is fine. However, I understand that it does not appear to be very mindful. Do you have another suggestion (also, generically for other such cases)?

Reply:
Server errors seem reasonable 🙂 As long as you know that it can fail and know what happens everything's good 👍